src/api.ts

```diff
@@ -3783,6 +3784,155 @@ export const methods = {
       countFailedResponse(api_name, 'exception in /contract/accesslist')
     }
   },
+  shardeum_getNodeList: async function (args: RequestParamsLike, callback: JSONRPCCallbackTypePlain) {
+    const api_name = 'shardeum_getNodeList'
+    const ticket = crypto
+      .createHash('sha1')
+      .update(api_name + Math.random() + Date.now())
+      .digest('hex')
+    logEventEmitter.emit('fn_start', ticket, api_name, performance.now())
+  
+    try {  
+      if (!Array.isArray(args) || args.length > 1 || (args.length === 1 && typeof args[0] !== 'object')) {
+        callback({ code: -32602, message: 'Invalid params: Expected an object or no parameters.' }, null);
+        countFailedResponse(api_name, 'Invalid params');
+        return;
+      }
+    
+      const params = args[0] || {};
+      const page = Math.max(1, parseInt(params.page) || 1);
+      const limit = Math.min(1000, Math.max(1, parseInt(params.limit) || 100));
+      const nodeListResult = await archiverAPI.getNodeList(page, limit);
+  
+      logEventEmitter.emit('fn_end', ticket, { success: true }, performance.now())
+      callback(null, nodeListResult);
+      countSuccessResponse(api_name, 'success', 'archiver');
+    } catch (error: any) {
+      logEventEmitter.emit('fn_end', ticket, { success: false, error: error.message }, performance.now())
+      callback(error, null);
+    }
+  },
+
+  shardeum_getNetworkAccount: async function (args: RequestParamsLike, callback: JSONRPCCallbackTypePlain) {
+    const api_name = 'shardeum_getNetworkAccount';
+    nestedCountersInstance.countEvent('endpoint', api_name);
+  
+    const ticket = crypto
+      .createHash('sha1')
+      .update(api_name + Math.random() + Date.now())
+      .digest('hex');
+  
+    logEventEmitter.emit('fn_start', ticket, api_name, performance.now());
+  
+    try {
+      const networkAccountData = await archiverAPI.getNetworkAccount();
+  
+      if (networkAccountData && networkAccountData.networkAccount && networkAccountData.networkAccount.data) {
+        const current = networkAccountData.networkAccount.data.current;
+        const archiver = current.archiver || {};
+  
+        // transform the response to match the expected structure
+        const response = {
+          activeVersion: archiver.activeVersion || current.activeVersion,
+          latestVersion: archiver.latestVersion || current.latestVersion,
+          minVersion: archiver.minVersion || current.minVersion,
+          certCycleDuration: current.certCycleDuration,
+          maintenanceFee: current.maintenanceFee,
+          maintenanceInterval: current.maintenanceInterval,
+          penalty: {
+            amount: current.nodePenaltyUsd.value,
+            currency: "shm"
+          },
+          slashing: {
+            leftNetworkEarlyPenaltyPercent: 0.2,
+            nodeRefutedPenaltyPercent: 0.2,
+            syncTimeoutPenaltyPercent: 0.2
```

Comment:
is it fine to have these values hardcoded?

Reply:
AFAIK yes as the values do not change